pkg/controller/lifecycle/actuator.go

```diff
@@ -703,14 +705,47 @@ func (a *actuator) deleteManagedDNSEntries(ctx context.Context, ex *extensionsv1
 				return nil
 			}
 		}
+		details := a.collectProviderDetailsOnDeletingDNSEntries(ctx, list)
 		return &reconcilerutils.RequeueAfterError{
-			Cause:        fmt.Errorf("waiting until shoot DNS entries have been deleted"),
+			Cause:        fmt.Errorf("waiting until shoot DNS entries have been deleted: %s", details),
 			RequeueAfter: 15 * time.Second,
 		}
 	}
 	return nil
 }
 
+func (a *actuator) collectProviderDetailsOnDeletingDNSEntries(ctx context.Context, list []dnsv1alpha1.DNSEntry) string {
+	providers := sets.NewString()
+	for _, item := range list {
+		if item.DeletionTimestamp != nil && item.DeletionTimestamp.Time.Add(1*time.Minute).Before(time.Now()) {
+			if item.Status.Provider != nil {
+				providers.Insert(*item.Status.Provider)
+			} else {
+				providers.Insert("")
+			}
+		}
+	}
+	var status []string
+	for k := range providers {
+		if k == "" {
+			status = append(status, "no suitable provider")
+			continue
+		}
+		parts := strings.Split(k, "/")
```

Comment:
Nit
```suggestion
		parts := strings.Split(k, types.Separator)
```

Package: k8s.io/apimachinery/pkg/types

You could also use `kutils.ParseObjectName` ([ref](https://github.com/gardener/gardener/blob/8ead4a3d7b30e67bd8f4c6a1041c9cb45fbe195b/pkg/utils/kubernetes/object.go#L46)).